src/App.tsx

```diff
@@ -1,60 +1,121 @@
+import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
 
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
-import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
-import { UserSelector } from './components/UserSelector';
+import { PostsList } from './components/PostsList/PostsList';
+import { PostDetails } from './components/PostDetails/PostDetails';
+import { UserSelector } from './components/UserSelector/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { getAllUsers, getPosts } from './api';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [openedPost, setOpenedPost] = useState<Post | null>(null);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  const shouldShowNoPostsMessage =
+    !errorMessage && !isLoading && selectedUser && !posts.length;
 
-              <Loader />
+  useEffect(() => {
+    getAllUsers()
+      .then(setUsers)
+      .catch(() => {
+        setErrorMessage('Something went wrong!');
+      });
+  }, []);
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  useEffect(() => {
+    if (!selectedUser) {
+      return;
+    }
+
+    setIsLoading(true);
+    getPosts(selectedUser?.id)
+      .then(setPosts)
+      .catch(() => {
+        setErrorMessage('Something went wrong!');
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, [selectedUser]);
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+  const handleSelectUser = (newUser: User) => {
```

Comment:
The method name `handleSelectUser` follows the recommended naming conventions for event handler functions in React.